contracts/StablecoinConverter.sol

```diff
@@ -0,0 +1,119 @@
+pragma solidity ^0.5.0;
+
+import "./EpochTokenLocker.sol";
+import "./libraries/IdToAddressBiMap.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+import "./libraries/IdToAddressBiMap.sol";
+
+
+contract StablecoinConverter is EpochTokenLocker {
+    using SafeMath for uint;
+    using BytesLib for bytes32;
+
+    event OrderPlacement(
+        address owner,
+        uint16 buyToken,
+        uint16 sellToken,
+        bool sellOrderFlag,
+        uint32 validFrom,
+        uint32 validUntil,
+        uint256 buyAmount,
+        uint256 sellAmount
+    );
+
+    event OrderCancelation(
+        address owner,
+        uint id
+    );
+
+    //outstanding volume of an order is encoded as buyAmount or sellAmount depending on sellOrderFlag
+    struct Order {
+        uint16 buyToken;
+        uint16 sellToken;
+        uint32 validFrom;  // order is valid from validFrom inclusively
+        uint32 validUntil;  // order is valid till validUntil inclusively
+        bool sellOrderFlag;
+        uint128 buyAmount;
+        uint128 sellAmount;
+    }
+
+    // User-> Order
+    mapping(address => Order[]) public orders;
+
+    IdToAddressBiMap.Data private registeredTokens;
+    uint public MAX_TOKENS;
```

Comment:
If it doesn't change it's value, it's more efficient to be declared as a constant.
Also, technically right now is not a constant, so it shouldn't be declared in capital letters. But I do think it should be a constant, so it's ok to keep it capitalized :) 

Regarding the order, constants are usually declared at the top, one possible order to follow in the declaration can be this one: https://github.com/0xcert/solidity-style-guide#order-of-layout

Reply:
It's variable so that the behavior of only allowing MAX_TOKENS can be easily unit tested. Otherwise it would be hard to achieve 100% coverage, which atm is a test passing criteria.

Reply:
Yeah, I had already a forth and back with Felix on how to unit test. We had to make a compromise, and the compromise we decided for was to not use constants, instead of writing weird subclassing contracts for testing.